src/pages/index.js

```diff
@@ -34,50 +36,126 @@ const enableValidation = (formSettings) => {
   nameSelector: ".profile__name",
   jobSelector: ".profile__job",
+  avatarSelector: ".profile__picture",
 });
 
 /* ********************** */
 /* Declarations of popups */
 /* ********************** */
+const avatarForm = new PopupWithForm("form[name='avatar-form']", (data) => {
+  avatarForm.submitButton.textContent = "Saving...";
+  api
+    .editProfilePicture(data.link)
+    .then((res) => (res.ok ? true : res.status))
+    .then(() => {
+      userInfo.setAvatar(data.link);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      avatarForm.submitButton.textContent = "Save";
+      avatarForm.close();
+    });
+});
+avatarForm.setEventListeners();
+
 const cardForm = new PopupWithForm("form[name='card-form']", (data) => {
-  cardGallery.addItem(data);
-  cardForm.close();
+  cardForm.submitButton.textContent = "Saving...";
+  api
+    .addNewCard({ name: data.name, link: data.link })
+    .then((res) => (res.ok ? res.json() : res.status))
+    .then((res) => {
+      cardGallery.addItem(res);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      cardForm.submitButton.textContent = "Create";
+      cardForm.close();
+    });
 });
 cardForm.setEventListeners();
 
 const profileForm = new PopupWithForm("form[name='profile-form']", (data) => {
-  userInfo.setUserInfo(data);
-  profileForm.close();
+  profileForm.submitButton.textContent = "Saving...";
+  api
+    .editUserInfo({ name: data.name, job: data.job })
+    .then((res) => (res.ok ? true : res.status))
+    .then(() => {
+      userInfo.setUserInfo({ name: data.name, about: data.job });
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      profileForm.submitButton.textContent = "Save";
+      profileForm.close();
+    });
 });
 profileForm.setEventListeners();
 
+const confirmDeletePopup = new PopupDeleteConfirmation(
+  "form[name='delete-form']"
+);
+confirmDeletePopup.setEventListeners();
+
 const picturePopup = new PopupWithImage(".popup__picture");
 picturePopup.setEventListeners();
 
 /* *********************************** */
 /* Declaration of card gallery section */
 /* *********************************** */
-const cardGallery = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const card = new Card(data, ".card-template", (data) => {
-        picturePopup.open(data);
+const cardGallery = new Section((data) => {
+  const card = new Card(
+    data,
+    ".card-template",
+    (data) => {
+      picturePopup.open(data);
+    },
+    () => {
+      confirmDeletePopup.setSubmitAction(() => {
+        api.deleteCard(data._id);
+        card.deleteCard();
+        confirmDeletePopup.close();
       });
-      const cardElement = card.generateCard();
-      cardGallery.container.prepend(cardElement);
+      confirmDeletePopup.open();
     },
-  },
-  ".gallery__cards"
-);
-
-cardGallery.renderItems();
+    () => {
+      const cardLikeIcon = card.card.querySelector(".card__like");
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards